### PR TITLE
Adaptations to handle PIV BER-TLV

### PR DESCRIPTION
**Pull request by NicsTr**
In [the NIST standard for Personal Identity Verification (PIV)](https://nvlpubs.nist.gov/nistpubs/specialpublications/nist.sp.800-73-4.pdf), BER-TLV structures are used. However, the standard specifies the following:

>The PIV data objects are BER-TLV objects encoded as per [ISO8825], except that tag values of the PIV data object’s inner tag assignments do not conform to BER-TLV requirements.13 This is due to the need to accommodate legacy tags inherited from [GSC-IS].

In particular, it means that it exists structures where the TLV should be of type primitive but the associated tag have the "constructed bit" set, eg. the GUID field of the CHUID structure having `0x34` as tag but being primitive (Table 9, page 40 of [NIST SP 800-73-4](https://nvlpubs.nist.gov/nistpubs/specialpublications/nist.sp.800-73-4.pdf)).

In order to not restart a library from scratch and use this crate to manipulate TLV in the context of PIV, I implemented a new opt-in feature: `piv`. This feature basically disables the consistency check of the tag's constructed bit with the actual value passed while constructing a TLV object. Not being able to tell if a TLV is constructed or primitive makes the recursive parsing impossible. Thus, I chose to only implement the parsing of a structure with maximum depth two. I used conditional compilation extensively to avoid modifying the behaviour or interfaces of the library when the `piv` feature is not enabled.

Since I am fairly new to Rust programming, let me know if it can be merged in this crate or if you see a better approach to solve this problem.

### CODE REVIEW

**Comment by jkowalsk** [comment]
Hi,
I'm glad my code could be usefull, and I will review and merge this. Nevertheless i'm currently on holliday and i can't commit myself to do this before january.
Thanks for this MR anyway, i'll keep you informed

**Comment by NicsTr** [comment]
Thanks for your quick answer. I'm not in a rush to have this merged, January is fine by me.

Have a good holiday.

**Comment by jkowalsk** [comment]
Hi,
I took time to take a look at your code but I quite don't like the feature usage.
It make code hard to follow.

As a replacement I quickly tried to define a GenericTag trait which can help managing your case.
Take a look at the "tag_trait" branch and the examples/piv.rs example 

I'm open to suggestions for improving this

**Comment by NicsTr** [comment]
Thanks a lot for your reactivity and work. I will try to use this branch for my project and keep you posted on how it goes.

**Comment by NicsTr** [comment]
I can confirm that it works great for my use case. I'm closing this MR. Thanks again.
